File: fintrack-frontend/src/app/core/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { UserResponse, UpdateProfileRequest, ChangePasswordRequest } from '../../models/user.model';

@Injectable({ providedIn: 'root' })
export class UserService {
  private apiUrl = 'http://localhost:8080/api/users';

  constructor(
    private http: HttpClient) {}

  getProfile(): Observable<UserResponse> {
    return this.http.get<UserResponse>(`${this.apiUrl}/me`);
  }

  updateProfile(request: UpdateProfileRequest): Observable<UserResponse> {
    return this.http.put<UserResponse>(`${this.apiUrl}/me`, request);
  }

  changePassword(request: ChangePasswordRequest): Observable<{ message: string }> {
    return this.http.put<{ message: string }>(`${this.apiUrl}/me/password`, request);
  }
}

File: fintrack-backend/src/main/java/com/fintrack/backend/service/UserService.java
package com.fintrack.backend.service;

import com.fintrack.backend.dto.ChangePasswordRequest;
import com.fintrack.backend.dto.UpdateProfileRequest;
import com.fintrack.backend.dto.UserResponseDTO;

public interface UserService {
    UserResponseDTO getProfile(String email);
    UserResponseDTO updateProfile(String email, UpdateProfileRequest request);
    void changePassword(String email, ChangePasswordRequest request);
}
